### PR TITLE
feat(gatsby-transformer-csv): add tsv parsing option

### PR DESCRIPTION
**Pull request by chooban**
# Description

Allow TSV files to be transformed using the `gatsby-transformer-csv` plugin. The existing code filters out any non-csv files, so passing the delimiter directly would not work. 

### Documentation

I've added a section to the readme.

## Related Issues

No related issues that I know of. I found myself with a TSV file, thought "I bet I can make the plugin handle this" and thought someone else might find it useful. Totally understand if the Gatsby community is not crying out for tsv transforming!

### CODE REVIEW

**Comment by pvdz** [comment]
I'm almost inclined to merge it but wonder why we wouldn't just propagate the delimiter directly? Why would we bother with an arbitrary character like tab (even when it might be common) when we could just propagate the delimiter option directly?

Is it very useful to do the csv/tsv dance? I'm not so sure.

cc @pieh for input

**Comment by chooban** [comment]
You can currently pass the tab character through directly, but [this line](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-transformer-csv/src/gatsby-node.js#L25) will then ignore any files that *aren't* of type `.csv`. I thought that adding the tab to the delimiters within the plugin might also be nicer than people wondering "How do I put a tab in there?"

From my pov, this isn't blocking me at all. It was a learning experience that I thought I'd offer back. I completely get the fact it'd then be more to maintain. :)

**Comment by pieh** [comment]
The `delimiter` is currently not documented in plugin's README (but it works as we pass options to `csvtojson`). My vote here would be:
 - add way to handle more extensions than just .csv (i.e. ``extensions: [`csv`, `tsv`]`` plugin option which would default to current `csv` as to not change default behaviour - might be worthy adding some comment to consider changing defaults for next major version release of the plugin)
 - document new plugin option (`extensions`) and also add documentation for hidden option: `delimiter` - possibly reusing what `csvtojson` have in its docs:
    > delimiter: delimiter used for separating columns. Use "auto" if delimiter is unknown in advance, in this case, delimiter will be auto-detected (by best attempt). Use an array to give a list of potential delimiters e.g. [",","|","$"]. default: ","
 - add recipe specific for `.tsv` to README (sample config using `extensions` and `delimiter`)

This way this plugin will be flexible (i.e. will be able to support more variants of "csv"-esque formats), and also have exact docs on how to get `.tsv` to work quickly

**Comment by chooban** [comment]
SGTM. I'll try and get this updated and back to you quickly.

**Comment by chooban** [comment]
I've added the `extensions` option and fleshed out the documentation around the possible options. I haven't copied the docs from csv2json as it would be duplication. If it changed in future there could be a mismatch. Unlikely, but possible.

I've merged my remote so that it's more to up to date with master. I know the recent blog post detailed that you're using rebasing for some changes, but I wasn't sure if that's the case for external PRs.

File: packages/gatsby-transformer-csv/src/gatsby-node.js [hunk 1]
```diff
@@ -49,7 +52,9 @@ async function onCreateNode(
   function transformObject(obj, i) {
     const csvNode = {
       ...obj,
-      id: obj.id ? obj.id : createNodeId(`${node.id} [${i}] >>> CSV`),
+      id: obj.id
+        ? obj.id
```
Comment (pvdz) [inline packages/gatsby-transformer-csv/src/gatsby-node.js]: This can be `??` as well (if you're going to change this line anyways :) )

Comment (chooban) [reply packages/gatsby-transformer-csv/src/gatsby-node.js]: I'm cool with that. Consistency is nice.

**Comment by pvdz** [comment]
>  I wasn't sure if that's the case for external PRs.

Yeah all PRs get squashed and rebased against master before merging

File: packages/gatsby-transformer-csv/README.md [hunk 2]
```diff
@@ -32,6 +32,10 @@ Above is the minimal configuration required to begin working. Additional
 customization of the parsing process is possible using the parameters listed in
 [csvtojson](https://github.com/Keyang/node-csvtojson#parameters).
 
+### Configuration
+
+For example, to pass the `noheader` option, you can configure like so:
```
Comment (laurieontech) [inline packages/gatsby-transformer-csv/README.md]:
```suggestion
For example, to pass the `noheader` option, you can configure the plugin in `gatsby-config.js` as follows:
```

**Comment by chooban** [comment]
@laurieontech AFAICS, by far the most options come from the csvtojson library, which is linked to, but could be more prominent. Since I've added an option that's not from that library I could try and make that clearer. 

How about...

======

### Configuration 

The example above is the minimal configuration required to begin working. Additional
customization of the parsing process is possible using the parameters listed in
[csvtojson](https://github.com/Keyang/node-csvtojson#parameters). Any parameter listed on that page can be passed directly to the library using the plugin options.

For example, the `noheader` option...

======

Does that draw attention to the other page a bit more?

**Comment by laurieontech** [comment]
@chooban that looks great. Let's go ahead and add that language.

**Comment by chooban** [comment]
You're welcome!

**Pull request merged**